test/orange/orangeinp/CsgTreeUtils.test.cc

```diff
@@ -65,6 +68,27 @@ class CsgTreeUtilsTest : public ::celeritas::test::Test
         return eval_sense(n);
     }
 
+    DeMorganSimplifierResult
+    validate_simplification(std::string_view expected,
+                            std::vector<NodeId> const& expected_node_map)
+    {
+        auto [simplified, node_map] = transform_negated_joins(tree_);
+        EXPECT_EQ(expected, to_string(simplified));
+        EXPECT_VEC_EQ(expected_node_map, node_map);
```

Comment:
The downside of `EXPECT_EQ` in a function is that the failure messages point here instead of the calling location where you want them...